examples/pendulum_swing_up_variable_duration.py

```diff
@@ -0,0 +1,120 @@
+"""This solves the simple pendulum swing up problem presented here:
+
+http://hmc.csuohio.edu/resources/human-motion-seminar-jan-23-2014
+
+A simple pendulum is controlled by a torque at its joint. The goal is to swing
+the pendulum from its rest equilibrium to a target angle by minimizing the
+energy used to do so in a minimal amount of time.
+
+Solves the same problem as in ``pendulum_swing_up.py`` but with a variable
+total duration.
+
+"""
+
+from collections import OrderedDict
+
+import numpy as np
+import sympy as sym
+from opty.direct_collocation import Problem
+from opty.utils import building_docs
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+
+target_angle = np.pi
+num_nodes = 500
+save_animation = False
+
+# Symbolic equations of motion
+m, g, d, t, h = sym.symbols('m, g, d, t, h')
+theta, omega, T = sym.symbols('theta, omega, T', cls=sym.Function)
+
+state_symbols = (theta(t), omega(t))
+constant_symbols = (m, g, d)
+specified_symbols = (T(t),)
+
+eom = sym.Matrix([theta(t).diff() - omega(t),
+                  m*d**2*omega(t).diff() + m*g*d*sym.sin(theta(t)) - T(t)])
+
+# Specify the known system parameters.
+par_map = OrderedDict()
+par_map[m] = 1.0
+par_map[g] = 9.81
+par_map[d] = 1.0
+
+
+# Specify the objective function and it's gradient.
+def obj(free):
```

Comment:
Why not use the `create_objective_function`?

Reply:
I wanted to write it out to see how the new variable affected the equations and did not know if `create_objective_function()` supported `h` as a symbol.

Reply:
Not yet but we can easily make it support it (will have to check ones more). But I think that the only difference is that we should change the default keyword argument to an integer (bit nicer if someone doesn't specify it) and we have to multiply the objective with the node_time_interval and remove it from the functions where it manually multiplies the solution with the node time interval.

Reply:
I'll let you update it after I merge this. Or I could merge in changes onto this branch if you want.